src/app/cli/src/coda_main.ml

```diff
@@ -597,7 +597,10 @@ struct
 
     (* TODO: This causes the signature to get checked twice as it is checked
    below before feeding it to add *)
-    let add t txn = apply_and_broadcast t [txn]
+    let add t txn =
+      apply_and_broadcast t
+        (Envelope.Incoming.wrap ~data:[txn]
+           ~sender:(Host_and_port.of_string "127.0.0.1:0"))
```

Comment:
Look at the file `Find_ip.ml`. Can you use this file to get a user's ip address rather than localhost? If finding their IP fails, can you log the error? I think `add` should be cancelled if the sender cannot input their actual IP address. If failing this way is too problematic, then you can default to sending the `Envelvoped` message with localhost.

Reply:
No, I'd rather not. These envelopes don't hit the wire, no other host will see these IPs. I want to use a distinct, usually-invalid `Host_and_port` pair here so that we can detect when we ourselves generate an invalid diff/transition.

Reply:
(the invalid part is port 0)